### ISSUE TITLE
perf: opportunistically execute next transaction batch

### ISSUE THREAD

**Issue by ryscheng** [oasislabs/oasis-core#105]
Currently, the compute node
1. Accumulates a batch
2. Requests the state
3. processes transaction batch
4. stores state to storage
5. commits to the blockchain
6. Responds to user.

In the future, we could begin executing the batch (n+1) before we hear back from consensus/storage on batch n. That way, the enclave is always doing work and never waiting on network requests.

**Comment by kostko** [comment]
Duplicate of #1581.
